src/main/java/org/broadinstitute/hellbender/tools/genome/SparkGenomeReadCounts.java

```diff
@@ -0,0 +1,134 @@
+        final long coverageCollectionEndTime = System.currentTimeMillis();
+        logger.info(String.format("Finished the spark coverage collection. Elapse of %d seconds",
+                (coverageCollectionEndTime - coverageCollectionStartTime) / 1000));
+
+        final List<SimpleInterval> fullGenomeBins = createFullGenomeBins(binsize, sequenceDictionary);
+        List<Target> fullGenomeTargetCollection = createTargetCollectionFromSimpleInterval(fullGenomeBins);
+        TargetCoverageUtils.writeTargetsAsBed(new File(outputFile.getAbsolutePath() + ".bed"), fullGenomeTargetCollection);
+
+        final Set<SimpleInterval> missingGenomeBins = Sets.difference( new HashSet<>(fullGenomeBins),byKey.keySet());
+        final Map<SimpleInterval, Long> byKeyMutable = new HashMap<>();
+        byKeyMutable.putAll(byKey);
+        missingGenomeBins.stream().forEach(m -> byKeyMutable.put(m, 0l));
+
+        final SortedMap<SimpleInterval, Long> byKeySorted = new TreeMap<>(IntervalUtils.LEXICOGRAPHICAL_ORDER_COMPARATOR);
+        byKeySorted.putAll(byKeyMutable);
+
+        TargetCoverageUtils.writeTargetsWithCoverageFromSimpleInterval(outputFile, sampleName, byKeySorted, comments);
+    }
+
+    public static List<SimpleInterval> createFullGenomeBins(final int binsize, final SAMSequenceDictionary sequenceDictionary){
+        final List<SimpleInterval> sequenceIntervals = IntervalUtils.getAllIntervalsForReference(sequenceDictionary);
+        return IntervalUtils.cutToShards(sequenceIntervals, binsize);
+    }
+
+    public static SimpleInterval createKey(final GATKRead read, final SAMSequenceDictionary sequenceDictionary){
+        final String contig = read.getContig();
+        final int newStart = (read.getStart()/binsize)*binsize+1;
+        final int newEnd = Math.min(newStart + binsize - 1, sequenceDictionary.getSequence(contig).getSequenceLength());
+
+        return new SimpleInterval(contig, newStart, newEnd);
+    }
+
+    @Override
+    protected void runTool(JavaSparkContext ctx) {
+        collectReads(ctx);
+    }
+
+    private void print(SortedMap<SimpleInterval, Long> byKeySorted, PrintStream ps) {
+        ps.println("CONTIG" + "\t" + "START" + "\t" + "END" + "\t" + "<ALL>");
+        for (final Locatable loc : byKeySorted.keySet()){
+            ps.println(loc.getContig() + "\t" + loc.getStart() + "\t" + loc.getEnd() + "\t" + byKeySorted.get(loc));
+        }
+    }
+
+    public ReadFilter makeReadFilter() {
+        return new WellformedReadFilter(getHeaderForReads())
+                .and(ReadFilterLibrary.MAPPED)
+                .and(ReadFilterLibrary.NOT_DUPLICATE)
+                .and(ReadFilterLibrary.NON_ZERO_REFERENCE_LENGTH_ALIGNMENT)
+                .and(ReadFilterLibrary.PASSES_VENDOR_QUALITY_CHECK);
+    }
+
+    public List<Target> createTargetCollectionFromSimpleInterval(final List<SimpleInterval> intervalList){
```

Comment:
Again.... public method needs javadoc or make it private. I suggest making it private.